tests/Unit/DataStructures/Test_Variables.cpp

```diff
@@ -788,21 +788,33 @@ void test_variables_assign_subset() noexcept {
               number_of_grid_points, values_in_variables.at(1)}));
 }
 
-SPECTRE_TEST_CASE("Unit.DataStructures.Variables.SliceVariables",
-                  "[DataStructures][Unit]") {
-  Variables<tmpl::list<VariablesTestTags_detail::vector>> vars(24, 0.);
-  const size_t x_extents = 2, y_extents = 3, z_extents = 4,
-               vec_size = VariablesTestTags_detail::vector::type::size();
+template <typename VectorType>
+void test_variables_slice() noexcept {
+  MAKE_GENERATOR(gen);
+  UniformCustomDistribution<size_t> sdist{5, 20};
```

Comment:
We should limit this one to `10` since we currently allow at most 12 grid points per dimension :)